### PR TITLE
all values are fine for value args, even if they don't make sense

### PR DESCRIPTION
**Pull request by gregsdennis**
Updates a few test scenarios per https://github.com/ietf-wg-jsonpath/draft-ietf-jsonpath-base/issues/431

### CODE REVIEW

File: cts.json [hunk 1]
```diff
@@ -2239,14 +2240,16 @@
...
-      "invalid_selector": true
+      "document" : [{"a": "bc"}],
+      "result": []
     },
     {
-      "name": "filter, match function, invalid second arg",
+      "name": "filter, match function, non-string second arg",
       "selector" : "$[?match(@.a, 1)]",
-      "invalid_selector": true
+      "document" : [{"a": "bc"}],
```
Comment (f3ath) [inline cts.json:2251]: Change to `[{"a": 1}]` maybe?

File: cts.json [hunk 2]
```diff
@@ -2320,14 +2323,16 @@
...
+      "result": []
     },
     {
-      "name": "filter, search function, invalid second arg",
+      "name": "filter, search function, non-string second arg",
       "selector" : "$[?search(@.a, 1)]",
-      "invalid_selector": true
+      "document" : [{"a": "bc"}],
```
Comment (f3ath) [inline cts.json:2334]: Same suggestion as above

File: cts.json (hunk 1 above)
Comment (gregsdennis) [reply cts.json:2251]: Why?  The point is that it doesn't select the item because `1` is not a string.

Comment (gregsdennis) [reply cts.json:2251]: But the expression isn't governed by the underlying engine.  It's our own syntax, and we don't have type coercion.

Also for the type system as it currently is, `1` and `"1"` are two unrelated values within the same type: `ValueType`.

Comment (gregsdennis) [reply cts.json:2251]: I think it would be better to have a test for `1=="1"` to test for that specifically.

Comment (gregsdennis) [reply cts.json:2251]: The second case only applies when there is coercion, and the tests I added in #23 ensure that doesn't happen.  It's not necessary to check that here.

Comment (glyn) [reply cts.json:2251]: Some general comments:

* Each test should aim to test one thing.

* We may be able to borrow some tests from the JSONPath comparison project, or at least get inspiration from there, particularly for tests that check that non-standard behaviour does not occur. (@cburgmer may care to comment.)

**Review by glyn — APPROVED** [review]
LGTM, thanks.

## Meta comment

Please note that I don't have sufficient bandwidth to give the CTS much attention. I'm delighted that there are now at least two others involved. (I'm still concerned about the lack of a good approach to https://github.com/jsonpath-standard/jsonpath-compliance-test-suite/issues/9, which I think needs implementing for the CTS to be sufficient. Maybe we can take up discussion of that in the issue.)

**Pull request merged**